lib/srv/desktop/windows_server.go

```diff
@@ -438,72 +438,85 @@ func (s *WindowsService) Serve(plainLis net.Listener) error {
 				return nil
 			}
 			return trace.Wrap(err)
+
+		}
+		proxyConn, ok := conn.(*tls.Conn)
+		if !ok {
+			return trace.ConnectionProblem(nil, "Got %T from TLS listener, expected *tls.Conn", conn)
 		}
 
-		go s.handleConnection(conn)
+		go s.handleConnection(proxyConn)
 	}
 }
 
 // handleConnection handles TLS connections from a Teleport proxy.
 // It authenticates and authorizes the connection, and then begins
 // translating the TDP messages from the proxy into native RDP.
-func (s *WindowsService) handleConnection(con net.Conn) {
-	defer con.Close()
+func (s *WindowsService) handleConnection(proxyConn *tls.Conn) {
+	defer proxyConn.Close()
 	log := s.cfg.Log
+	tdpConn := tdp.NewConn(proxyConn)
+
+	// logFn should be something like s.cfg.Log.WithError(err).Warning or s.cfg.Log.Error
+	logAndSendTdpError := func(message string, logFn func(args ...interface{})) {
```

Comment:
You're never using `args` here, so kind of defeating the purpose of variadic args.